sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorBuilder.java

```diff
@@ -0,0 +1,183 @@
+ * checkpoints after successfully processing every event, configure the following fields:
+ *
+ * <ul>
+ * <li>Consumer group name.</li>
+ * <li>{@link EventHubAsyncClient} - An asynchronous Event Hub client the {@link EventProcessor} will use for
+ * consuming events.</li>
+ * <li>{@link #processEvent(Function)} - A lambda function that will be called when new events are received.</li>
+ * </ul>
+ *
+ * <p>
+ * <strong>Creating an {@link EventProcessor} using a {@link #processEvent(Function)} function</strong>
+ * </p>
+ * {@codesnippet com.azure.messaging.eventhubs.eventprocessorbuilder.processevent}
+ *
+ * <p>
+ *  To create a more advanced {@link EventProcessor} that allows you to control the creation of
+ *  {@link PartitionProcessor}s and lets you control the frequency of checkpointing, configure the
+ *  {@link PartitionProcessorFactory} instead of {@link #processEvent(Function)} lambda function.
+ * </p>
+ *
+ * <p>
+ * <strong>Creating an {@link EventProcessor} using {@link PartitionProcessorFactory}</strong>
+ * </p>
+ * {@codesnippet com.azure.messaging.eventhubs.eventprocessorbuilder.partitionprocessorfactory}
+ *
+ * @see EventProcessor
+ * @see EventHubConsumer
+ */
+public class EventProcessorBuilder {
+    private final ClientLogger logger = new ClientLogger(EventProcessorBuilder.class);
+
+    private EventPosition initialEventPosition;
+    private PartitionProcessorFactory partitionProcessorFactory;
+    private String consumerGroup;
+    private PartitionManager partitionManager;
+    private EventHubAsyncClient eventHubAsyncClient;
+    private Function<EventData, Mono<Void>> processEvent;
+
+    /**
+     * Sets the Event Hub client the {@link EventProcessor} will use to connect to the Event Hub for consuming events.
+     *
+     * @param eventHubAsyncClient The Event Hub asynchronous client for consuming events.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder eventHubClient(EventHubAsyncClient eventHubAsyncClient) {
+        this.eventHubAsyncClient = eventHubAsyncClient;
+        return this;
+    }
+
+    /**
+     * Sets the consumer group name from which the {@link EventProcessor} should consume events.
+     *
+     * @param consumerGroup The consumer group name this {@link EventProcessor} should consume events.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder consumerGroup(String consumerGroup) {
+        this.consumerGroup = consumerGroup;
+        return this;
+    }
+
+    /**
+     * Sets the lambda function that will be used to process events from all partitions this {@link EventProcessor} is
+     * responsible for.
+     *
+     * @param processEvent The lambda function that will be used to process events.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder processEvent(Function<EventData, Mono<Void>> processEvent) {
```

Comment:
Do you need `Function<EventData, Mono<Void>>`? You don't document what the user is supposed to actually return, and I am not sure you have a use case for it anyway? Could it be `Consumer<EventData>` instead?